### PR TITLE
tokens: added tokens, colors, deprecations

### LINKED ISSUE #2243: tokens: Update to core palette
### Description

We're updating the core color tokens to adhere to our new palette to improve accessibility, flexibility, and brand expression. This includes updates to semantic token references. 

### Screenshots
image

### Resources

[Migration tables](https://www.figma.com/file/kRncCBxmHGoiSf5imuHNRM/Color-Migration-2024?type=design&node-id=11-23984&mode=design&t=XmqhbCRDXKma2Z2A-11)

[Token table](https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/eBay-Design-System?type=design&node-id=201652-6820&mode=design&t=4rtSEz8Nin7ttHHW-11)

### Engineering pre-requisites

- [X] Successful design review with Design System Team.
- [X] Supports dark mode.
- [X] Supports responsive design.
- [X] Tokens have been provided (or already exist).
- [ ] Successful accessibility review with Core A11Y Team.
- [ ] Successful frontend review with eBayUI Team.

### PR DESCRIPTION
**Pull request by ArtBlue**
Fixes #2243 

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
New tokens, color revisions, color family deprecations and removals.

## Screenshots

tokens

## Checklist

- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue

- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

File: src/sass/star-rating-select/star-rating-select.scss
Comment (agliga) [inline src/sass/star-rating-select/star-rating-select.scss:9]: Should we switch these to semantic tokens possibly?

**Review by agliga — COMMENTED** [review]
Overall quick pass through: looks good
I think just one question. I'm fine either way. 

Will try it locally later. Otherwise looks good to merge.

File: src/sass/star-rating-select/star-rating-select.scss
Comment (ArtBlue) [reply src/sass/star-rating-select/star-rating-select.scss:9]: There's a comment right above that regarding the decision to use the core token instead of the semantic token.

**Comment by agliga** [comment]
Approving, but lets wait for other PRs to merge before merging this.

**Review by agliga — APPROVED** [review]

**Comment by agliga** [comment]
I think we're good to merge this

**Pull request merged**
